File: prob1_bronze_season26contest1/java/TestRunner.java
import java.io.BufferedInputStream;
import java.io.IOException;
import java.io.PrintWriter;
import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class TestRunner {
    private static long solveCase(long a, long b, long cA, long cB, long fA) {
        long initialA = (b / cB) * cA + a;
        if (initialA >= fA) {
            return 0L;
        }

        long needA = fA - 1 - initialA;
        long y = cB - 1 - (b % cB);

        if (cA >= cB) {
            y += needA;
        } else {
            y += (needA / cA) * cB + (needA % cA);
        }

        return y + 1;
    }

    private static List<String> runFile(Path file) throws IOException {
        FastScanner fs = new FastScanner(file);
        int t = fs.nextInt();
        List<String> outputs = new ArrayList<>();
        if (t == -1) {
            return outputs;
        }
        for (int i = 0; i < t; i++) {
            long a = fs.nextLong();
            long b = fs.nextLong();
            long cA = fs.nextLong();
            long cB = fs.nextLong();
            long fA = fs.nextLong();
            outputs.add(Long.toString(solveCase(a, b, cA, cB, fA)));
        }
        return outputs;
    }

    public static void main(String[] args) throws Exception {
        Path testDir = Paths.get("..", "testData").normalize();
        if (!Files.isDirectory(testDir)) {
            System.out.println("testData folder not found: " + testDir.toAbsolutePath());
            return;
        }

        List<Path> files = new ArrayList<>();
        try (DirectoryStream<Path> stream = Files.newDirectoryStream(testDir, "*.in")) {
            for (Path p : stream) {
                files.add(p);
            }
        }
        files.sort(Comparator.comparingInt(p -> p.getFileName().toString().length())
                .thenComparing(p -> p.getFileName().toString()));

        PrintWriter out = new PrintWriter(System.out);
        for (Path p : files) {
            out.println(p.getFileName() + ":");
            for (String line : runFile(p)) {
                out.println(line);
            }
            out.println("-");
        }
        out.flush();
    }

    private static class FastScanner {
        private final BufferedInputStream in;
        private final byte[] buffer = new byte[1 << 16];
        private int ptr = 0;
        private int len = 0;

        FastScanner(Path path) throws IOException {
            in = new BufferedInputStream(Files.newInputStream(path));
        }

        private int read() throws IOException {
            if (ptr >= len) {
                len = in.read(buffer);
                ptr = 0;
                if (len <= 0) {
                    return -1;
                }
            }
            return buffer[ptr++];
        }

        int nextInt() throws IOException {
            int c;
            do {
                c = read();
                if (c == -1) {
                    return -1;
                }
            } while (c <= ' ');
            int sign = 1;
            if (c == '-') {
                sign = -1;
                c = read();
            }
            int val = 0;
            while (c > ' ') {
                val = val * 10 + (c - '0');
                c = read();
            }
            return val * sign;
        }

        long nextLong() throws IOException {
            int c;
            do {
                c = read();
                if (c == -1) {
                    return -1L;
                }
            } while (c <= ' ');
            int sign = 1;
            if (c == '-') {
                sign = -1;
                c = read();
            }
            long val = 0;
            while (c > ' ') {
                val = val * 10 + (c - '0');
                c = read();
            }
            return val * sign;
        }
    }
}

File: prob1_bronze_season26contest1/cpp/test_runner.cpp
#include <bits/stdc++.h>
#include <filesystem>
using namespace std;

using ll = long long;

ll solve_case(ll a, ll b, ll c_a, ll c_b, ll f_a) {
    ll initial_a = (b / c_b) * c_a + a;
    if (initial_a >= f_a) {
        return 0;
    }

    ll need_a = f_a - 1 - initial_a;
    ll y = c_b - 1 - (b % c_b);

    if (c_a >= c_b) {
        y += need_a;
    } else {
        y += (need_a / c_a) * c_b + (need_a % c_a);
    }

    return y + 1;
}

vector<string> run_file(const string &path) {
    ifstream in(path);
    vector<string> out;
    int t;
    if (!(in >> t)) {
        return out;
    }
    for (int i = 0; i < t; i++) {
        ll a, b, c_a, c_b, f_a;
        in >> a >> b >> c_a >> c_b >> f_a;
        out.push_back(to_string(solve_case(a, b, c_a, c_b, f_a)));
    }
    return out;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    string test_dir = "..\\testData";
    vector<string> files;

    for (const auto &entry : std::filesystem::directory_iterator(test_dir)) {
        if (!entry.is_regular_file()) {
            continue;
        }
        string name = entry.path().filename().string();
        if (name.size() >= 3 && name.substr(name.size() - 3) == ".in") {
            files.push_back(name);
        }
    }

    sort(files.begin(), files.end(), [](const string &a, const string &b) {
        if (a.size() != b.size()) {
            return a.size() < b.size();
        }
        return a < b;
    });

    for (const string &name : files) {
        string path = test_dir + "\\" + name;
        vector<string> outputs = run_file(path);
        cout << name << ":\n";
        for (const string &line : outputs) {
            cout << line << "\n";
        }
        cout << "-\n";
    }

    return 0;
}
